extras/gizmo/axis-shapes.js

```diff
@@ -0,0 +1,650 @@
+import {
+    createBox,
+    createCone,
+    createCylinder,
+    createPlane,
+    createMesh,
+    Color,
+    MeshInstance,
+    Entity,
+    Quat,
+    Vec3
+} from 'playcanvas';
+
+import { MeshTriData } from './mesh-tri-data.js';
+
+// constants
+const SHADOW_DAMP_SCALE = 0.25;
+const SHADOW_DAMP_OFFSET = 0.75;
+const TORUS_RENDER_SEGMENTS = 80;
+const TORUS_INTERSECT_SEGMENTS = 20;
+const LIGHT_DIR = new Vec3(1, 2, 3);
+const MESH_TEMPLATES = {
+    box: createBox,
+    cone: createCone,
+    cylinder: createCylinder,
+    plane: createPlane,
+    torus: createTorus
+};
+
+// temporary variables
+const tmpV1 = new Vec3();
+const tmpV2 = new Vec3();
+const tmpQ1 = new Quat();
+
+function createTorus(device, opts = {}) {
+    // Check the supplied options and provide defaults for unspecified ones
+    const rc = opts.tubeRadius ?? 0.2;
+    const rt = opts.ringRadius ?? 0.3;
+    const segments = opts.segments ?? 30;
+    const sides = opts.sides ?? 20;
+    const sectorAngle = opts.sectorAngle ?? 2 * Math.PI;
+
+    // Variable declarations
+    const positions = [];
+    const normals = [];
+    const uvs = [];
+    const indices = [];
+
+    for (let i = 0; i <= sides; i++) {
```

Comment:
this seems like a copy of createTorus in the engine .. do we need a copy of it, or can we just call what we have? I have not compared, but ideally we don't duplicate code like this.

Reply:
It is almost the same but this version allows for the sector size to be set. I was hoping this would replace the version inside the engine.

Reply:
feel free to just add the extra parameters you need to the one in the engine

Reply:
I was planning of leaving that for a follow up PR since this PR has no core engine changes